include/fast_float/ascii_number.h

```diff
@@ -74,9 +75,9 @@ uint32_t parse_eight_digits_unrolled(uint64_t val) {
   val = (((val & mask) * mul1) + (((val >> 16) & mask) * mul2)) >> 32;
   return uint32_t(val);
 }
-
+template <typename TCH>
 fastfloat_really_inline FASTFLOAT_CONSTEXPR20
-uint32_t parse_eight_digits_unrolled(const char *chars)  noexcept  {
+uint32_t parse_eight_digits_unrolled(TCH const * chars)  noexcept  {
```

Comment:
This routine expects ASCII/UTF-8. It will fail on UTF-8 inputs, for example.

Reply:
read_u64 handles the conversion of integer size between char types

Reply:
I standard corrected. However, it would still be be needed to check that it makes sense to do so: it is a very cheap check given ASCII inputs, but if you need to do more work, it may end up being a net negative. So we would need to benchmark with and without this optimization to make sure it is warranted.

Reply:
I understand, my reasoning for all of this is that I needed an unicode version, so I don't have to double buffer with fast or wcstod, now I'm going to add hex support using radix bases, and finally an ignore list, as for the tests I'll update them as I go, I guess I'll add a new enum 'infer' to chars_format to work as a pseudo drop-in replacement for wcstod behaviour.

Reply:
Sure. Great work. What you are doing is perfectly reasonable.

Ignore my initial negative and misguided comment.

Reply:
No worries, it's such a delicate change, and everywhere.

Reply:
> I stand corrected. However, it would still be be needed to check that it makes sense to do so: it is a very cheap check given ASCII inputs, but if you need to do more work, it may end up being a net negative. So we would need to benchmark with and without this optimization to make sure it is warranted.

I have a version of fast_float in which I've attempted to optimize for other char_types using opt-in SIMD. It also adds support for different parsing rules (JSON parsing rules, opt-in inf/nan etc.) --> if this is okay I'm happy to submit a PR or modify this one.